### PR TITLE
Week_3

### PR DESCRIPTION
**Pull request by Rom-T**

### CODE REVIEW

File: 02-express-tutorial/app.js [hunk 1]
```diff
@@ -1 +1,58 @@
...
+  if (!product) {
+    return res.status(404).json({ message: 'That product was not found' });
+  }
+
+  return res.json(product);
+});
+
+app.get('/api/v1/query', (req, res) => {
+  const { search, limit, priceLimit } = req.query;
+  let foundedProducts = [...products];
```
Comment (cagallo) [inline 02-express-tutorial/app.js]: Nice use of the spread operator! 

Is `founded` supposed to be` found`?

File: 02-express-tutorial/app.js [hunk 2]
```diff
@@ -1 +1,58 @@
...
+  }
+
+  if (limit) {
+    foundedProducts = foundedProducts.slice(0, parseInt(limit));
+  }
+
+  if (priceLimit) {
+    foundedProducts = foundedProducts.filter(product => {
+      return product.price <= parseInt(priceLimit);
+    });
```
Comment (cagallo) [inline 02-express-tutorial/app.js]: This is purely a suggestion but you could also use the implicit return of the function when you have simple return values.

`const foundedProducts = foundedProducts.filter((product) => product.price <= parseInt(priceLimit))`

This is from MDN on [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions):

```
Arrow functions can have either an expression body or the usual block body.

In an expression body, only a single expression is specified, which becomes the implicit return value. In a block body, you must use an explicit return statement.

JS
Copy to Clipboard
const func = (x) => x * x;
// expression body syntax, implied "return"

const func2 = (x, y) => {
  return x + y;
};
// with block body, explicit "return" needed
```

**Review by cagallo — APPROVED** [review]
Looks good!

File: 02-express-tutorial/app.js (hunk 1 above)
Comment (Rom-T) [reply 02-express-tutorial/app.js]: Yes, you are right. I'll fix it now. Thank you.

File: 02-express-tutorial/app.js (hunk 2 above)
Comment (Rom-T) [reply 02-express-tutorial/app.js]: I know about this type of return, but the second one is always easier for me to understand. But you are absolutely right - you need to relearn and use implicit return for simple return values.

**Pull request merged**